server/unittools.c

```diff
@@ -3357,6 +3357,12 @@ static void wakeup_neighbor_sentries(struct unit *punit)
           && can_unit_exist_at_tile(penemy, unit_tile(penemy))) {
         set_unit_activity(penemy, ACTIVITY_IDLE);
         send_unit_info(NULL, penemy);
+        notify_player(unit_owner(penemy), unit_tile(punit),
+                    E_UNIT_ORDERS, ftc_server,
+                    _("%s sentry wokeup after enemy (%s v%d %s| id: %d | hp: %d | pos: (%d, %d)) movement was "
```

Comment:
I'm trying to submit some info like this in https://www.hostedredmine.com/issues/913055
I'm using the format:`[id:137 A:1.0 HP:10]`
What about using this similar format here? Also once I get my patch in can use my code:
`unit_veteran_level_string()`.

Reply:
Yes. It would be nice to have similar formatting in all messages. But, I would prefer both attack and defence if we are reporting this. How do you want to do about it ? How about 
```
Unit sentry wokeup after vet_name unit_name [id:id A/D:A/D HP:hp] was spotted in tile (pos)
```

Reply:
The reason for keeping defense out of the upstream patch is that it can (in theory) leak intel that would otherwise be invisible to the attacker, like a small wonder that gives extra defense.

Reply:
Which small wonder gives extra defence to units separately ? But even if it does in some ruleset, Can we know that by looking at the defence probability in popup ? If we can, then it should be included here too imo. Because I think, if a player can notice when logged in then the sentry should inform him. 

Also, the same logic applies for attack. Some ruleset can give attack bonus too. We can probably do my initial way where both the attack and the defence isn't mentioned. But I think we should do both because of the popup argument.

Reply:
Something like the Great Wall but invisible... Don't know of any such wonder in the wild, but upstream maintainers see it as a breaking change that can't be done in a stable branch. The client shows what it knows, and maybe it's even smart enough to show [45%-55%] if there's the possibility in the ruleset.

I don't think there's a way to give attack bonuses in the ruleset.

Reply:
Then let us not include attack too ? If the attack can't be changed for now, it is better to leave with unit name and vet level in order to future proof it just in case? A player can anyways know the base attack from unit name and vet level.